Makefile

```diff
@@ -228,7 +228,8 @@ generate-kustomize-api: $(MYGOBIN)/k8scopy
 .PHONY: test-unit-kustomize-api
 test-unit-kustomize-api: build-kustomize-api
 	cd api; go test ./...  -ldflags "-X sigs.k8s.io/kustomize/api/provenance.version=v444.333.222"
-	cd api/krusty; OPENAPI_TEST=true go test -run TestCustomOpenAPIFieldFromComponentWithOverlays
+	cd api/krusty; OPENAPI_TEST=true go test -run TestOpenAPICustomFieldFromComponentWithOverlays
+	cd api/krusty; OPENAPI_TEST=true go test -p 1 -run TestCustomOpenApiField
```

Comment:
There is a global variable in `openapi.go` that stores the parsed schema. There are some race conditions when running these tests in parallel, so I separated them out.

Reply:
The argument to `-run` is a regex, right? Would it work to have a single `OPENAPI_TEST=true go test -p 1 -run OpenAPI -p 1` invocation?

Reply:
I tried that but for whatever reason it was still running into race conditions somehow. `TestOpenAPICustomFieldFromComponentWithOverlays` passes by itself, but fails if I try to run it with the other tests (which is why I changed the order of the words in the test so that `TestCustomOpenApiField` won't match it). I can try to figure out why but I think somehow it's still running things in parallel even with `-p 1`.

Reply:
Is it always that test that fails, never any of the others? That makes me wonder if it is a sequencing problem rather than concurrency, and there's a real problem with one or multiple tests mutating some state somehow.

I was actually able to confirm this by just literally joining together `TestCustomOpenApiFieldFromComponent` and `TestOpenAPICustomFieldFromComponentWithOverlays`. I get the same failure as I get from running the whole API suite together. I'm out of time for today, but I can help debug this tomorrow if you want. I'm worried both that we're papering over a real problem and that we'll end up not running some of our tests because of some silly "openAPI" vs "openApi" style problem causing the regex to not match.

Reply:
Thank you for verifying that it's a sequence issue, it helped me narrow down the problem. If the sequence matters, I would have expected `ResetOpenApi` to take care of it. Turns out it wasn't resetting everything, only some of the OpenAPI data.

I added that fix, and updated this make target accordingly to have a single `OPENAPI_TEST=true go test -p 1 -run OpenAPI -p 1` as you originally suggested.

Reply:
As discussed in meeting, I've separated out the openapi tests to their own package.